CodeEdit/NavigatorSidebar/ProjectNavigator/OutlineView/OutlineViewController.swift

```diff
@@ -0,0 +1,195 @@
+/// A `NSViewController` that handles the **ProjectNavigator** in the **NavigatorSideabr**.
+///
+/// Adds a ``outlineView`` inside a ``scrollView`` which shows the folder structure of the
+/// currently open project.
+class OutlineViewController: NSViewController {
+
+    var scrollView: NSScrollView!
+    var outlineView: NSOutlineView!
+
+    var content: [WorkspaceClient.FileItem] = []
+
+    var workspace: WorkspaceDocument?
+
+    var iconColor: AppPreferences.FileIconStyle = .color
+
+    /// Setup the ``scrollView`` and ``outlineView``
+    override func loadView() {
+        self.scrollView = NSScrollView()
+        self.view = scrollView
+
+        self.outlineView = NSOutlineView()
+        self.outlineView.delegate = self
+        self.outlineView.dataSource = self
+        self.outlineView.headerView = nil
+        self.outlineView.menu = OutlineMenu()
+        self.outlineView.menu?.delegate = self
+
+        let column = NSTableColumn(identifier: .init(rawValue: "Cell"))
+        column.title = "Cell"
+        outlineView.addTableColumn(column)
+
+        self.scrollView.documentView = outlineView
+        self.scrollView.contentView.automaticallyAdjustsContentInsets = false
+        self.scrollView.contentView.contentInsets = .init(top: 10, left: 0, bottom: 0, right: 0)
+        reloadContent()
+    }
+
+    init() {
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError()
+    }
+
+    /// Updates the selection of the ``outlineView`` whenever it changes.
+    ///
+    /// Most importantly when the `id` changes from an external view.
+    func updateSelection() {
+        guard let itemID = workspace?.selectionState.selectedId,
+              let item = try? workspace?.workspaceClient?.getFileItem(itemID) else { return }
+
+        let row = outlineView.row(forItem: item)
+        print(item.fileName, row)
+        outlineView.selectRowIndexes(.init(integer: row), byExtendingSelection: false)
+    }
+
+    /// Get the folder structure and store it into ``content``.
+    ///
+    /// Also creates a top level item "root" which represents the projects root directory and automatically expands it.
+    private func reloadContent() {
+        guard let folderURL = workspace?.workspaceClient?.folderURL() else { return }
+        let children = workspace?.selectionState.fileItems.sortItems(foldersOnTop: true)
+        let root = WorkspaceClient.FileItem(url: folderURL, children: children)
+        self.content = [root]
+        outlineView.reloadData()
+        guard let item = outlineView.item(atRow: 0) else { return }
+        outlineView.expandItem(item)
+    }
+
+    /// Get the appropriate color for the items icon depending on the users preferences.
+    /// - Parameter item: The `FileItem` to get the color for
+    /// - Returns: A `NSColor` for the given `FileItem`.
+    private func color(for item: WorkspaceClient.FileItem) -> NSColor {
+        if item.children == nil {
```

Comment:
I think this would be better:
```
if item.children == nil && iconColor == .color {
  return NSColor(item.iconColor)
} else {
  return .secondaryLabelColor
}
```